os/tools/mksyscall.c

```diff
@@ -482,15 +485,20 @@ static void generate_stub(int nparms)
 	/* Tail end of the function.  If the proxied function has no return
 	 * value, just return zero (OK).
 	 */
-
-	if (strcmp(get_parm(RETTYPE_INDEX), "void") == 0)
-		fprintf(stream, ");\n  return 0;\n}\n\n");
-	else
-		fprintf(stream, ");\n}\n\n");
-
-	cond_parm = get_parm(COND_INDEX);
-	if (cond_parm[0] != '\0')
-		fprintf(stream, "#endif /* %s */\n", cond_parm);
+	if(cond_parm[0] == '\0'){
+		if (strcmp(get_parm(RETTYPE_INDEX), "void") == 0)
+			fprintf(stream, ");\n  return 0;\n}\n\n");
+		else
+			fprintf(stream, ");\n}\n\n");
+	}
+	else{
+		fprintf(stream, ");\n");
+		fprintf(stream, "#else\n");
+    	fprintf(stream, "  set_errno(ENOSYS);\n");
+    	fprintf(stream, "  return -1;\n");
```

Comment:
We cannot return -1 always. It depends on the return type declared for each syscall. Please change as below.

```
if (strcmp(get_parm(RETTYPE_INDEX), "void") == 0)
    fprintf(stream, ");\n  return 0;\n}\n\n");
else if (strcmp(get_parm(RETTYPE_INDEX), "int") == 0)
    fprintf(stream, "  return -1;\n");
else 
    fprintf(stream, "  return NULL;\n");
```